### PR TITLE
Expand option manager test coverage

### PR DESCRIPTION
**Pull request by ahojnnes**
## Summary
- Expand test coverage for `controllers/option_manager.cc` from 13 tests to 51 tests
- Cover quality modifier methods (ModifyFor*), PostParse file loading, Write/Read round-trips for all option categories, INI parsing edge cases, idempotency, and command-line parsing for sub-options

## New test coverage areas
- **Quality modifiers** (7 tests): Verify all ModifyFor{Individual,Video,Internet}Data and ModifyFor{Low,Medium,High,Extreme}Quality set expected values
- **PostParse** (4 tests): Mapper image list, global mapper image list, constant rig/camera list file loading, and empty path handling
- **Write/Read round-trips** (12 tests): Bundle adjustment, stereo, meshing, render, gravity refiner, reconstruction clusterer, sequential/spatial/transitive pairing, two-view geometry, global mapper, and all-options round-trip
- **INI parsing** (2 tests): Unregistered options with allow_unregistered=true (warning path) and allow_unregistered=false (rejection path)
- **Idempotency** (1 test): All remaining Add*Options methods called twice without error
- **Command-line parsing** (4 tests): Feature matching, mapper sub-options, global mapper sub-options, patch match stereo
- **Other** (8 tests): Composite Add* dependency wiring, Reset-then-re-add, constructor with add_project_options=false, Check on defaults, ResetOptions for all sub-structs, Read failure on Check failure, cumulative quality modifiers

## Test plan
```
cd build && cmake --build . --target colmap_controllers_option_manager_test -j$(sysctl -n hw.logicalcpu)
./src/colmap/controllers/option_manager_test
```

All 51 tests pass:
```
[==========] Running 51 tests from 1 test suite.
[  PASSED  ] 51 tests.
```

### CODE REVIEW

File: src/colmap/controllers/option_manager_test.cc [hunk 1]
```diff
@@ -338,15 +353,1015 @@ TEST(OptionManager, ParseUnknownArgumentsFails) {
...
+  EXPECT_EQ(options.feature_extraction->max_image_size, 1600);
+  EXPECT_EQ(options.feature_extraction->sift->max_num_features, 4096);
+  EXPECT_EQ(options.mapper->ba_global_max_refinements, 2);
+  EXPECT_EQ(options.patch_match_stereo->max_image_size, 1600);
+  EXPECT_EQ(options.patch_match_stereo->window_radius, 4);
+  EXPECT_EQ(options.patch_match_stereo->window_step, 2);
+  EXPECT_EQ(options.patch_match_stereo->num_iterations, 5);
+  EXPECT_FALSE(options.patch_match_stereo->geom_consistency);
+  EXPECT_EQ(options.stereo_fusion->max_image_size, 1600);
+}
```
[the remark this answers is not in the document]
Comment (ahojnnes) [reply src/colmap/controllers/option_manager_test.cc:457]: Keeping tests self-contained for clarity. Each test's setup documents its specific configuration.

File: src/colmap/controllers/option_manager_test.cc [hunk 2]
```diff
@@ -338,15 +353,1015 @@ TEST(OptionManager, ParseUnknownArgumentsFails) {
...
+  options.bundle_adjustment->refine_focal_length = false;
+
+  options.ResetOptions(/*reset_paths=*/false);
+
+  EXPECT_EQ(options.patch_match_stereo->max_image_size,
+            pm_defaults.max_image_size);
+  EXPECT_NE(options.stereo_fusion->min_num_pixels, 99);
+  EXPECT_NE(options.poisson_meshing->depth, 42);
+  EXPECT_TRUE(options.bundle_adjustment->refine_focal_length);
+}
```
[the remark this answers is not in the document]
Comment (ahojnnes) [reply src/colmap/controllers/option_manager_test.cc:1145]: Same as above - keeping each test self-contained for readability.
